lib/elixir/lib/system.ex

```diff
@@ -5,40 +5,71 @@ defmodule System do
   with the VM or the host system.
   """
 
-  defp strip_re(iodata, pattern) do
-    :re.replace(iodata, pattern, "", [return: :binary])
+  @base_dir           Path.join([__DIR__, "..", "..", ".."])
+  @version_file       Path.join(@base_dir, "VERSION")
+  @git_dir            Path.join(@base_dir, ".git")
+  @git_head_file      Path.join(@git_dir, "HEAD")
+
+  @spec strip(iodata) :: String.t
+  defp strip(iodata) do
+    iodata
+    |> :re.replace("^[\s\r\n\t]+", "", [return: :binary])
+    |> :re.replace("[\s\r\n\t]+$", "", [return: :binary])
```

Comment:
previous version did not work for anything that was not a white spaces (that is. tabs, and carriage returns),
and neither it work when spaces were both at the beginning and the end of the file.
 
I cannot make it work squashing it into a single Regex.
```elixir
iex> "\nstrip this\n" |> :re.replace("^[\s\r\n\t]+|[\s\r\n\t]+$", "", [return: :binary])
"strip this\n"
```
but I'm sure someone with more knowledge of erlang can do it :-)